multiple_events_reader.C

```diff
@@ -0,0 +1,119 @@
+//max_r and max_z are obtained from find_max_r_and_z.C
+#define max_r 156.763
+#define max_z 137.928
+
+R__LOAD_LIBRARY(/opt/WCSim/build/install/lib/libWCSimRoot.so)
+
+int multiple_events_reader(const char* filename="/work/kmtsui/wcte/cosmic/hk_flux/wcsim_mu-_*.root") {
+
+    //serve for case checking
+    int losing_track = 0, track_ignored = 0;
+
+    //Get the files
+    TChain* t = new TChain("wcsimT");
+    t->Add(filename);
+
+    //setup of histogram
+    TH2D* entrance_display = new TH2D("Entrance_position","Entrance position;[cm];[cm]",300,-TMath::Pi()*max_r,TMath::Pi()*max_r,300,-max_z-2*max_r,max_z+2*max_r);
+    TH2D* exit_display = new TH2D("Exit_position","Exit position;[cm];[cm]",300,-TMath::Pi()*max_r,TMath::Pi()*max_r,300,-max_z-2*max_r+5,max_z+2*max_r+5);
+    TH1D* Energy_dissipation = new TH1D("Energy_dissipation", "Energy dissipation distribution;[MeV/cm];Number of event", 200, 0, 10);
+    TH2D* Q_vs_E = new TH2D("Q_vs_E", "Total charge against energy loss;Energy dissipation [Mev];Total charge;", 3000, 0, 15000, 10000, 0, 40000);
+    double barrelCut = max_z-0.1;
+    
+    int nEvent = t->GetEntries();
+    std::cout << "Total number of event in all the given files: " << nEvent << "\n\n";
+    WCSimRootEvent* wcsimrootsuperevent = new WCSimRootEvent();
+    t->SetBranchAddress("wcsimrootevent", &wcsimrootsuperevent);
+    WCSimRootTrack* wcsimroottrack = new WCSimRootTrack();
+
+    for (int i=0; i<nEvent; i++) {
+
+        //progress
+        std::cout << "\rReading event " << i << "/" << nEvent;
```

Comment:
Better not to print every events, maybe do it for e.g. every 1 percent of all events